exporter/datadogexporter/internal/metrics/transform.go

```diff
@@ -0,0 +1,241 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/metrics"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/metricstransformprocessor"
+)
+
+// TransformFunc takes a set of metrics and transforms them.
+type TransformFunc func(ctx context.Context, md pmetric.Metrics) error
+
+// NewInfraTransformFunc returns a new TransformFunc which converts OpenTelemetry system and process metrics
+// to Datadog compatible ones.
+func NewInfraTransformFunc(ctx context.Context, set component.ExporterCreateSettings) (TransformFunc, error) {
+	cid := config.NewComponentIDWithName(config.MetricsDataType, "dd-infra-metricstransformer")
+	tcfg := &metricstransformprocessor.Config{
+		ProcessorSettings: config.NewProcessorSettings(cid),
+		Transforms:        metricTransforms,
+	}
+	pset := component.ProcessorCreateSettings(set)
+	noop, err := consumer.NewMetrics(
+		consumer.ConsumeMetricsFunc(
+			func(_ context.Context, _ pmetric.Metrics) error { return nil },
+		),
+	)
+	if err != nil {
+		return nil, err
+	}
+	proc, err := metricstransformprocessor.NewFactory().CreateMetricsProcessor(ctx, pset, tcfg, noop)
+	if err != nil {
+		return nil, err
+	}
+	return proc.ConsumeMetrics, err
+}
+
+// metricTransforms defines a set of metric transformations for converting OpenTelemetry
+// "hostmetrics" to Datadog-compatible infrastructure metrics.
+var metricTransforms = []metricstransformprocessor.Transform{
```

Comment:
Should we consider driving this via a config file (file within datadog exporter) . I think we might need to expand this to add more metrics like kubernetes ones etc . Having in a file would be easier to expand . 

This is more of open question rather than strong opinion .

Reply:
I initially wanted to have a separate yaml file and use the [embed](https://pkg.go.dev/embed) package, but that's not supported in the Go version that the Collector is currently using. Then I considered importing an [external package](https://github.com/shurcooL/vfsgen) to generate a virtual file system. I ultimately decided it was overkill and that having the rendered config is the easiest because we don't concern ourselves with parsing yaml, potential errors, and should be equally easy to configure.

What do you think? Do you have any other ideas I've not considered? Why do you think that extending this structure would be more difficult than writing yaml? They are basically the same thing.

Reply:
Done.